infrastructure/learn/service.go

```diff
@@ -204,13 +204,14 @@ func (s *serviceImpl) updateCaches(lessons []Lesson) {
 			lessonsForRule = append(lessonsForRule, lesson)
 			s.lessonsByRuleCache.Set(lesson.Slug, lessonsForRule, expiration)
 		}
-
-		lessonsForEcosystem, exists := s.lessonsByEcosystemCache.Get(lesson.Ecosystem)
-		if !exists {
-			lessonsForEcosystem = []Lesson{}
+		for _, ecosystem := range lesson.Ecosystems {
+			lessonsForEcosystem, exists := s.lessonsByEcosystemCache.Get(ecosystem)
+			if !exists {
+				lessonsForEcosystem = []Lesson{}
+			}
+			lessonsForEcosystem = append(lessonsForEcosystem, lesson)
+			s.lessonsByEcosystemCache.Set(ecosystem, lessonsForEcosystem, expiration)
 		}
-		lessonsForEcosystem = append(lessonsForEcosystem, lesson)
-		s.lessonsByEcosystemCache.Set(lesson.Ecosystem, lessonsForEcosystem, expiration)
 	}
```

Comment:
I'm not sure if it's related to this change or it's a random issue but I observed the following:
|main|fix/snyk-learn-for-all-ecosystems|
|-|-|
|![main-with-link](https://github.com/snyk/snyk-ls/assets/1948377/0b0a6693-fc11-4140-b81a-b7f6d91b4787)|![missing-link](https://github.com/snyk/snyk-ls/assets/1948377/94eb7c0a-916f-47df-b894-9d24c6f83aca)|

Reply:
Either a regression or it was wrong before. I'll have a look!

Reply:
Reproduced it, it was a bug before - it was linking to the Path Traversal learn lesson, which it shouldn't have.